src/QirRuntime/test/QIR-static/qir-test-math.cpp

```diff
@@ -23,3 +27,56 @@ TEST_CASE("QIR: Math.ArcTan2", "[qir.math][qir.Math.ArcTan2]")
     REQUIRE(0 == Microsoft__Quantum__Testing__QIR__Math__ArcTan2Test__body());
 }
 
+TEST_CASE("QIR: Math.DrawRandomInt", "[qir.math][qir.Math.DrawRandomInt]")
+{
+    // Commented out until https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r573978601
+    // is resolved.
+
+    /*
+    // Range of the generated random numbers:
+    constexpr int64_t   minimum = -1024;
+    constexpr int64_t   maximum =  1023;
+    constexpr size_t    count =  maximum - minimum + 1;     // How many numbers are in the range.
+
+    size_t genAverage = 1000;                               // Each number should be generated ~1000 times.
+    size_t times = count * genAverage;                      // How many times to draw.
+
+    std::vector<uint16_t>  counters(count, 0);              // How many times each random number has been generated.
+
+    // Count how many times each random number has been generated.
+    while(--times)
+    {
+        const int64_t randomNumber = 
+            Microsoft__Quantum__Testing__QIR__Math__TestDrawRandomInt__body(minimum, maximum);
+        REQUIRE(minimum <= randomNumber);
+        REQUIRE(randomNumber <= maximum);
+
+        // Increment the counter for this number:
+        const int64_t index = randomNumber - minimum;
+        REQUIRE(0 <= index);
```

Comment:
Isn't it the same as `REQUIRE(minimum <= randomNumber)` above?

Reply:
Yes, it is (currently).

However in the future the values of the `minimum` and `maximum` can change, the mechanism of mapping them to the `index` can also change.  
By double-checking the `index` I want to make sure that the container is indexed correctly, no matter what.